arbnode/espresso_snapshot_handler.go

```diff
@@ -0,0 +1,95 @@
+package arbnode
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/espresso/authdb"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+type EspressoSnapshotHandler struct {
+	stopwaiter.StopWaiter
+	db               *authdb.AuthDB
+	parentChainDir   string
+	l2chainDataDir   string
+	useSnapshot      bool
+	generateSnapshot bool
+}
+
+func NewEspressoSnapshotHandler(db *authdb.AuthDB, parentChainDir string, l2chainDataDir string, useSnapshot bool, generateSnapshot bool) *EspressoSnapshotHandler {
+	return &EspressoSnapshotHandler{
+		db:               db,
+		parentChainDir:   parentChainDir,
+		l2chainDataDir:   l2chainDataDir,
+		useSnapshot:      useSnapshot,
+		generateSnapshot: generateSnapshot,
+	}
+}
+
+func (s *EspressoSnapshotHandler) StoreSnapshotSha256(sum string) error {
+	path := filepath.Join(s.parentChainDir, "snapshot.txt")
```

Comment:
1. should indicate the block number and block hash in the name, i'd use `SNAPSHOT_<num>_<hash>` as file name
2. should we store the raw bytes instead of a string here?